polyfactory/factories/base.py

```diff
@@ -661,6 +664,34 @@ def get_model_fields(cls) -> list[FieldMeta]:  # pragma: no cover
         """
         raise NotImplementedError
 
+    @classmethod
+    def get_factory_fields(cls) -> list[tuple[str, Any]]:
+        factory_fields = cls.__dict__.items()
+        return [
+            (field_name, field_value) for field_name, field_value in factory_fields
+            if not (field_name.startswith("__") or field_name == "_abc_impl")
+        ]
+
+
+    @classmethod
+    def _check_declared_fields_exist_in_model(cls):
+        model_fields_names = [field_meta.name for field_meta in cls.get_model_fields()]
+        factory_fields = cls.get_factory_fields()
+
+        for field_name, field_value in factory_fields:
+            if field_name in model_fields_names or inspect.isfunction(field_value):
```

Comment:
Why is this `inspect.isfunction` needed?

Reply:
This is done to completely ignore any method or class method on the factory as we currently don’t have any way to identify them…

Reply:
Aah yeah this was in combination with the other discussion regarding the warnings on the class level attributes itself.